modules/serverless-event-delivery-component-behaviors.adoc

```diff
@@ -0,0 +1,26 @@
+// Module included in the following assemblies:
+//
+// serverless/event_workflows/serverless-event-delivery.adoc
+
+[id="serverless-event-delivery-component-behaviors_{context}"]
+= Event delivery behavior for Knative Eventing components
+
+Different Knative Eventing components have their own behavior patterns that are followed for event delivery.
+
+== Delivery failure status codes
+
+The channel or broker receiver may respond with the following status codes if an event fails to be delivered:
+
+`500`:: This is a generic status code which means that the event was not delivered successfully.
+`404`:: This status code means that the channel or broker the event is being delivered to does not exist, or that the `Host` header is incorrect.
+`400`:: This status code means that the event being sent to the receiver is invalid.
+
+== Default channel and broker event delivery
+
+Developers can set event delivery parameters in the subscriber configuration to ensure that any events that fail to be delivered from `InMemoryChannel` channels or the `default` Knative broker dispatchers to an event sink are retried. You must also configure the dead letter sink for the subscriber if you want to provide a sink where events that are not eventually delivered can be stored, otherwise undelivered events are dropped.
+
+== Apache Kafka channel and broker event delivery
+
+If an event is successfully delivered to a Kafka channel or broker receiver, the receiver will respond with a 202 status code, which means that the event has been safely stored inside a Kafka topic and will not be lost. If the receiver responds with any other status code, the event has not been safely stored, and steps must be taken by the user to resolve this issue.
+
+Developers can set event delivery parameters in the subscriber configuration to ensure that any events that fail to be delivered from the Kafka channel dispatcher to an event sink are retried. You must also configure the dead letter sink for the subscriber if you want to provide a sink where events that are not eventually delivered can be stored, otherwise undelivered events are dropped.
```

Comment:
This is duplicated from above, and as we talk here about 'common behaviour/patterns' I wouldn't repeat it here as this behaviour is already caught by the previous section.